src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -173,7 +173,6 @@ void symbol_factoryt::gen_nondet_array_init(
   const auto &size = array_type.size();
   PRECONDITION(size.id() == ID_constant);
   auto const array_size = numeric_cast_v<size_t>(to_constant_expr(size));
-  DATA_INVARIANT(array_size > 0, "Arrays should have positive size");
```

Comment:
Does this code assume that the type of `size` is an unsigned type?

Reply:
Thank you for calling this out! I have now reworked this bit such as neither to rely on unsigned values nor on the compilation platform's `size_t` limitation.